thrust/thrust/system/cuda/detail/core/agent_launcher.h

```diff
@@ -31,22 +31,37 @@
 #include <cub/detail/device_synchronize.cuh>
 
 #if THRUST_DEVICE_COMPILER == THRUST_DEVICE_COMPILER_NVCC
-#include <thrust/system/cuda/detail/guarded_cuda_runtime_api.h>
 #include <thrust/system/cuda/detail/core/triple_chevron_launch.h>
 #include <thrust/system/cuda/detail/core/util.h>
+#include <thrust/system/cuda/detail/guarded_cuda_runtime_api.h>
+
 #include <cassert>
 
 #include <nv/target>
 
+#include <cuda/std/detail/__config>
+
 THRUST_NAMESPACE_BEGIN
 namespace cuda_cub {
 namespace core {
 
+/**
+ * @def THRUST_DISABLE_KERNEL_VISIBILITY_WARNING_SUPPRESSION
+ * If defined, the default suppression of kernel visibility attribute warning is disabled.
+ */
+#if !defined(THRUST_DISABLE_KERNEL_VISIBILITY_WARNING_SUPPRESSION)
+_LIBCUDACXX_GCC_DIAGNOSTIC_IGNORED("-Wattributes")
+_LIBCUDACXX_CLANG_DIAGNOSTIC_IGNORED("-Wattributes")                      
+#endif
+
+#ifndef THRUST_DETAIL_KERNEL_ATTRIBUTES
+#define THRUST_DETAIL_KERNEL_ATTRIBUTES __global__ _LIBCUDACXX_HIDDEN
+#endif
 
 #if defined(__CUDA_ARCH__) || defined(_NVHPC_CUDA)
 #if 0
   template <class Agent, class... Args>
-  void __global__
+  void THRUST_DETAIL_KERNEL_ATTRIBUTES
```

Comment:
Nitpick:

In cub you prefixed with `THRUST_DETAIL_KERNEL_ATTRIBUTES` here you put it after the return type